### ISSUE TITLE
Postprocessing of derivative quantities

### ISSUE THREAD

**Issue by roystgnr** [grinsfem/grins#289]
Paraview doesn't give us the best results for vorticities or heat fluxes; we should add them as optional postprocessing variables.

**Comment by pbauman** [comment]
Yes, this is quickly moving up my todo list (I need stress postprocessing). My plan was to add two new options to postprocessing: "element quantities" and (node) "patch recovery". The former would be just averaging over quadrature points to get the element value and output as element (cell in ParaView parlance) quantities. (Paraview has CellToPoint filters that could give nodal fields, where it averages the cell values to get the nodal value.) The latter is doing Zienkiewicz-Zhu.

Anything else you want added, would rather see, or want as a priority here?

Can't give any concrete timeline for when I'll actually get to this other than I need this since yesterday.

**Comment by roystgnr** [comment]
Should we add a Zienkiewicz-Zhu option to libMesh system_projection.C ?  Right now that's what we use to put a PostProcessedQuantities function on the grid, but it's implicitly assuming that if you've got a C0 FEM space then what you're putting on it is also C0.

The "element quantities" option ought to be fairly easy by contrast; just change GRINS to (optionally) use a DG space in PostProcessedQuantities::initialize() and libMesh will get the rest right.

**Comment by pbauman** [comment]
> Should we add a Zienkiewicz-Zhu option to libMesh system_projection.C ? Right now that's what we use to put a PostProcessedQuantities function on the grid, but it's implicitly assuming that if you've got a C0 FEM space then what you're putting on it is also C0.

Up to you. I was planning to make it a part of the grins postprocessing, but if you think that's a worthwhile feature in libMesh, we can target that instead. I guess we add a Z-Z flag or something?

> The "element quantities" option ought to be fairly easy by contrast; just change GRINS to (optionally) use a DG space in PostProcessedQuantities::initialize() and libMesh will get the rest right.

Oh sweet, that's way easier than what I was thinking.

**Comment by nicholasmalaya** [comment]
Just chiming in to say I would absolutely love this capability, would be happy to be a guinea pig or assist in any way with this.
